test/coverage.overlay

```diff
@@ -0,0 +1,167 @@
+module "postgrest-7.0.1-inplace/PostgREST.ApiRequest" {
+  inside "userApiRequest" {
+    tick "otherwise" on line 291;
+    tick "Nothing" on line 291;
```

Comment:
How will we maintain/generate this file? It seems that if any lines change, we need to update this?

Reply:
Yes, this is something that I don't know about, yet. I figured we'd need to try and see. It seems like this could become ugly... :o).

This seems to be a challenge in the whole hpc overlay concept. I'm not experienced enough in using it to see what we can do about it, yet.

Reply:
Looks quite messy, would impact pretty much every commit. Leaving it out would mean that we get a lower coverage ratio and less transparency on what is actually testable, right?

Reply:
Yes. I'm not that much concerned about coverage ratio, but much more about the mental overhead of deciding "should this be tested or not?" every time we look at the coverage reports. This will quickly lead to mistakes, where we don't test something that we should.

I thought a bit more about it - and the overlay tool is only really usable, once we are at 100% coverage. If we're at that point we can just do an automatic `postgrest-coverage-draft-overlay` for each commit and get the proper file, because everything left is "untestable". We can then tell easily from the diff of the overlay file whether just line numbers were changed or whether something was added that should actually be tested.

WDYT?

Reply:
Yes, that makes sense! It will take some expertise to interpret the coverage reports in the meantime.

Not sure that we should aim for 100% coverage at all cost... Let's see how far we get with reasonable effort.

Reply:
Ok, so we will merge this without the overlay file first.

Maybe we should not go for 100% "at all cost". But everything that we decide the cost is too high, we can add to the overlay file, too, because we decided against testing it. In that sense we should absolutely go for 100% (test+overlay), because this is the only scenario that will be easily manageable without too much mental overhead.